### DISCUSSION TITLE
Re-validating on fix

### DISCUSSION THREAD

**Discussion by sgoodluck** [houseform/houseform#87] [Q&A] [answered]
I have a form with several fields all of which should only be validated on formSubmit.
If a field has an error, the formSubmit displays the errors as expected. 

ISSUE

If I correct the issue listed, the state of the form is still `!valid`
If my submit button is disabled when `!valid` there is no way to resubmit. 

I'd love some insight here!

**Comment by crutchcorn** [comment, marked as answer]
Which version of HouseForm are you using? Can you provide a reproduction? We ran into this previously with an older version of HouseForm but it was patched.

Edit: See @joaom00's reply below for the actual answer

**Reply by sgoodluck** [reply]
HouseForm Version
1.9.0

React Version
18.2.0

For reproduction -- do you have a preference for what that looks like?

**Reply by sgoodluck** [reply]
Not sure if this is related (it might be) -- but currently digging into an issue tracked [here](https://github.com/houseform/houseform/issues/88). We are using pnpm and this is a NextJS project if that's relevant.

**Reply by sgoodluck** [reply]
Standby... creating a minimal repro

**Reply by sgoodluck** [reply]
Created here: https://github.com/sgoodluck/houseform-repro-issue-88

I can confirm that it repros and that it may be the source of the submit issue mentioned here

**Reply by joaom00** [reply]
@sgoodluck You also need to use `onChangeValidate` to update `isValid` while the user is typing.

You want something like: "validate first on submit after on change".

To do this, you can rely on the `isSubmitted` boolean. An example https://stackblitz.com/edit/houseform-v1-example-rfsp1k?file=App.tsx

**Reply by sgoodluck** [reply]
Thank you for the example -- that is extremely clear!
